### PR TITLE
Refactoring IrcClient

### PR DESCRIPTION
**Pull request by Denrage**

### CODE REVIEW

**Review by Hueppy — APPROVED** [review]
lgtm

File: src/Projects/IrcClient/Clients/DccClient.cs [hunk 1]
```diff
@@ -25,7 +25,8 @@ public DccClient(string filename, ulong filesize, int buffersize = 4096)
 
         public event Action<ulong, ulong> Progress;
 
-        public async Task Connect(string host, int port)
+        // TODO: use CancellationToken for socket when it's implemented in .net
+        public async Task Connect(string host, int port, CancellationToken token)
```
Comment (Phili9000) [inline src/Projects/IrcClient/Clients/DccClient.cs:29]: Dotnet 5 has an overload with CancellationToken I think, so when/if we move to dotnet 5, we can change this

File: src/Projects/IrcClient/Clients/DccClient.cs [hunk 2]
```diff
@@ -25,7 +25,8 @@ public DccClient(string filename, ulong filesize, int buffersize = 4096)
...
         {
             await this.socket.ConnectAsync(host, port).ConfigureAwait(false);
         }
```
Comment (Phili9000) [inline src/Projects/IrcClient/Clients/DccClient.cs:32]: Dotnet 5 has an overload with CancellationToken I think. So when/if we move to dotnet 5, we could change this easily

**Review by Phili9000 — APPROVED** [review]

File: src/Projects/IrcClient/Clients/DccClient.cs (hunk 2 above)
Comment (Denrage) [reply src/Projects/IrcClient/Clients/DccClient.cs:32]: We saw the extension class in source where it got implemented but why so ever it is not in the released version dotnet even if it got implemented multiple month ago like here https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketTaskExtensions.cs
But yeah the todo is there so we revisit this in the future

**Pull request merged**